src/components/Menu/index.js

```diff
@@ -1,49 +1,27 @@
-import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { modularizeParentCss } from '../../higherOrderComponents';
+import propsMapper from './propsMapper';
 import CompositeMenu from './CompositeMenu';
-import { mapProps, modularize } from '../../utils/component';
 
-function buildItemTree(items) {
-  const root = { id: 0, depth: -1, children: [] };
-  const parents = [root];
-
-  _.forEach(items, (item, index) => {
-    const parent = (() => {
-      while (_.last(parents).depth >= item.depth) {
-        parents.pop();
-      }
-      return _.last(parents);
-    })();
-
-    const itemWithChildren = { ...item, children: [] };
-    parent.children.push(itemWithChildren);
-
-    if (item === _.last(items)) {
-      return;
-    }
-
-    if (items[index + 1].depth > item.depth) {
-      delete itemWithChildren.href;
-      parents.push(itemWithChildren);
-    }
-  });
-
-  return root;
+class Menu extends React.PureComponent {
```

Comment:
What do you think about composing HOCs instead of define class?

```js
const propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    menu_title: PropTypes.string,
    menu_url: PropTypes.string,
    is_newtab: PropTypes.bool,
    menu_deep: PropTypes.number,
  })),
};

export default compose(
  modularizeParentCss(),
  setPropTypes(propTypes),
  mapProps(propsMapper),
)(CompositeMenu);
```